src/cpprealm/app.hpp

```diff
@@ -280,10 +388,11 @@ class App {
 
         m_app = app::App::get_shared_app(app::App::Config{
             .app_id=app_id,
-            .transport = std::make_shared<DefaultTransport>(),
+            .transport = std::make_shared<internal::DefaultTransport>(),
+            .base_url = base_url.empty() ? util::Optional<std::string>() : base_url,
```

Comment:
why not just make `baas_url` an optional param to begin with?

Reply:
Good call.